src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -95,6 +95,20 @@ class LinalgBackendBase
 	#undef BACKEND_GENERIC_DOT
 
 	/**
+	* Wrapper method that computes mean of SGVectors and SGMatrices.
+	*
+	* @see linalg::mean
+	*/
+	#define BACKEND_GENERIC_MEAN(Type, Container) \
+	virtual Type mean(const Container<Type>& a) const \
```

Comment:
so are we sure that the mean of a `SGVector<int32_t>` should be expressed as an `int32_t` and not at least a `float32_t`? because that's not really a mean...